src/ca.cpp

```diff
@@ -336,15 +336,14 @@ int create_capmt(SCAPMT *ca, int listmgmt, uint8_t *capmt, int capmt_len,
     int pos = 0;
     SPMT *pmt = get_pmt(ca->pmt_id);
     SPMT *other = get_pmt(ca->other_id);
-    int version = ca->version;
 
     if (!pmt)
         LOG_AND_RETURN(1, "%s: PMT %d not found", __FUNCTION__, ca->pmt_id);
 
     capmt[pos++] = listmgmt;
     copy16(capmt, pos, ca->sid);
     pos += 2;
-    capmt[pos++] = ((version & 0xF) << 1) | 0x1;
+    capmt[pos++] = ca->version;
```

Comment:
The version is also shifted to the left 1 bit: https://github.com/openpli-arm/dvb-apps/blob/17a557bce03c9cf61d0d3b29c4c387b84f95196b/lib/libdvben50221/en50221_app_ca.c#L240

Reply:
I need to revisit that and check, could be I was looking at the whole byte value only without realizing it containa current/next indicator too.

Reply:
Fixed